src/optimizerinterface.jl

```diff
@@ -30,15 +30,30 @@ function MOIU.attachoptimizer!(model::Model)
 end
 
 
-function optimize(model::Model;
-                ignore_optimize_hook=(model.optimizehook===nothing))
+"""
+    function optimize(model::Model,
+                      factory::Union{Nothing, OptimizerFactory} = nothing;
```

Comment:
It's better to call it `optimizer_factory` (in code) and "optimizer factory" (in text). `factory` is too generic.